homeassistant/components/tradfri/light.py

```diff
@@ -297,11 +293,3 @@ def _refresh(self, device):
         # Caching of LightControl and light object
         self._device_control = device.light_control
         self._device_data = device.light_control.lights[0]
-        self._features = SUPPORTED_FEATURES
-
-        if device.light_control.can_set_dimmer:
```

Comment:
Actually, I think we need this in the `_refresh` method. The color picker doesn't show up when I removed this code.

Reply:
And I also get this unless the code is as in the latest push:
```
Traceback (most recent call last):
  File "/home/patrik/Repos/hass/home-assistant/homeassistant/helpers/entity_platform.py", line 245, in async_add_entities
    for entity in new_entities
  File "/home/patrik/Repos/hass/home-assistant/homeassistant/helpers/entity_platform.py", line 242, in <listcomp>
    self._async_add_entity(
  File "/home/patrik/Repos/hass/home-assistant/homeassistant/components/tradfri/light.py", line 42, in <genexpr>
    async_add_entities(TradfriLight(light, api, gateway_id) for light in lights)
  File "/home/patrik/Repos/hass/home-assistant/homeassistant/components/tradfri/light.py", line 148, in __init__
    super().__init__(device, api, gateway_id)
  File "/home/patrik/Repos/hass/home-assistant/homeassistant/components/tradfri/base_class.py", line 27, in __init__
    self._refresh(device)
  File "/home/patrik/Repos/hass/home-assistant/homeassistant/components/tradfri/light.py", line 298, in _refresh
    self._features |= SUPPORT_BRIGHTNESS
AttributeError: 'TradfriLight' object has no attribute '_features'
```

Reply:
Ok yeah we need to reset the features before updating it in refresh.

It's just that the features should never change during the entity lifetime, so it's a hit weird that we have this in the refresh method. Can we calculate it before the first update somehow?

Reply:
I refactored the code a bit. Is this an approved pattern to use?